pkg/storage/etcd/etcd_helper.go

```diff
@@ -558,6 +576,15 @@ func (h *etcdHelper) GuaranteedUpdate(ctx context.Context, key string, ptrToType
 			ttl = *newTTL
 		}
 
+		// Since update object may have a resourceVersion set, we need to clear it here.
+		if h.versioner != nil {
+			if err := h.versioner.UpdateObject(ret, meta.Expiration, 0); err != nil {
+				return errors.New("resourceVersion cannot be set on objects store in etcd")
+			}
+		} else {
+			return errors.New("couldn't access version without versioner")
```

Comment:
@smarterclayton @lavalamp - if we can do this, I will fix some failing tests; If you think that assuming versioner is incorrect, what do you think we should do?